manifests/job.pp

```diff
@@ -43,6 +44,9 @@
       jobname => $jobname,
     }
   } else {
+    if $config == undef and $source == undef and $template == undef {
+      fail('You should at least specify one of the $config, $source or $template param')
+    }
```

Comment:
Why is this `if` statement (which is obviously useful and avoids misleading configuration) making your rspec tests fail ? Sorry I'm not really used to making rspec tests so I think I will not be able to modify them the right way